subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/IncrementalCompilerFactory.java

```diff
@@ -49,41 +50,39 @@
 
     private final FileOperations fileOperations;
     private final StreamHasher streamHasher;
-    private final FileHasher fileHasher;
     private final GeneralCompileCaches generalCompileCaches;
     private final BuildOperationExecutor buildOperationExecutor;
     private final StringInterner interner;
     private final FileSystemSnapshotter fileSystemSnapshotter;
+    private final FileHasher fileHasher;
 
-    public IncrementalCompilerFactory(FileOperations fileOperations, StreamHasher streamHasher, FileHasher fileHasher, GeneralCompileCaches generalCompileCaches, BuildOperationExecutor buildOperationExecutor, StringInterner interner, FileSystemSnapshotter fileSystemSnapshotter) {
+    public IncrementalCompilerFactory(FileOperations fileOperations, StreamHasher streamHasher, GeneralCompileCaches generalCompileCaches, BuildOperationExecutor buildOperationExecutor, StringInterner interner, FileSystemSnapshotter fileSystemSnapshotter, FileHasher fileHasher) {
         this.fileOperations = fileOperations;
         this.streamHasher = streamHasher;
-        this.fileHasher = fileHasher;
         this.generalCompileCaches = generalCompileCaches;
         this.buildOperationExecutor = buildOperationExecutor;
         this.interner = interner;
         this.fileSystemSnapshotter = fileSystemSnapshotter;
+        this.fileHasher = fileHasher;
```

Comment:
Nit: there's no need to change the argument order of the constructor, is there?

Reply:
No, this was removed and later added again :) But it doesn't hurt either, so I'm gonna spare our CI another round.